render-ws-java-client/src/test/java/org/janelia/render/client/parameter/StackIdWithZParametersTest.java

```diff
@@ -0,0 +1,127 @@
+package org.janelia.render.client.parameter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.janelia.alignment.spec.stack.StackId;
+import org.janelia.alignment.spec.stack.StackIdNamingGroup;
+import org.janelia.render.client.RenderDataClient;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests the {@link StackIdWithZParameters} class.
+ *
+ * @author Eric Trautman
+ */
+public class StackIdWithZParametersTest {
+
+    private final String owner = "testOwner";
+    private final String projectA = "testProjectA";
+    private final String projectB = "testProjectB";
+
+    private final List<StackId> stackIds = Arrays.asList(
+            new StackId(owner, projectA, "stack1"),
+            new StackId(owner, projectA, "stack2"),
+            new StackId(owner, projectA, "stack1_align"),
+            new StackId(owner, projectA, "stack2_align"),
+            new StackId(owner, projectA, "stack1_align_ic"),
+            new StackId(owner, projectA, "stack2_align_ic"),
+            new StackId(owner, projectB, "stack1"),
+            new StackId(owner, projectB, "stack2"),
+            new StackId(owner, projectB, "stack1_align"),
+            new StackId(owner, projectB, "stack2_align"),
+            new StackId(owner, projectB, "stack1_align_ic"),
+            new StackId(owner, projectB, "stack2_align_ic")
+    );
+
+    private final RenderDataClient mockDataClient = new MockRenderDataClient(owner, projectA, stackIds);
+
+    @Test
+    public void testGetEligibleStackIds() throws IOException {
```

Comment:
It seems to me that this doesn't really test the behavior of `StackIdWithZParameters`, but just tests if the line
```Java
return hasProjectPattern ? renderDataClient.getOwnerStacks() : renderDataClient.getProjectStacks();
```
does the right thing (which I would classify as implementation detail and most of which is mocked anyway).

Was `getEligibleStackIds` only made public to be able to test it here? If so, it should probably remain private and not be tested at all.

Reply:
Yeah - I drafted this originally because I hadn't found the flipped logic bug in StackIdNamingGroup yet.  I agree it is easier to drop the test and restore getEligibleStackIds to a private method (and rely on testGetStackIdList to catch future issues).